userapi/src/main/java/fk/prof/userapi/UserapiConfigManager.java

```diff
@@ -61,73 +43,54 @@ public static void setDefaultSystemProperties() {
-    }
-    return s3Config;
+  Configuration.S3Config getS3Config() {
+    return getStorageConfig().s3Config;
   }
 
-
-  JsonObject getStorageThreadPoolConfig() {
-    JsonObject tpConfig = getStorageConfig().getJsonObject(THREAD_POOL);
-    checkNotEmpty(tpConfig, "thread pool");
-    return tpConfig;
+  Configuration.FixedSizeThreadPoolConfig getStorageThreadPoolConfig() {
+    return getStorageConfig().tpConfig;
   }
 
-  JsonObject getStorageConfig() {
-    JsonObject storageConfig = config.getJsonObject(STORAGE);
-    checkNotEmpty(storageConfig, "storage");
-    return storageConfig;
-  }
+  Configuration.StorageConfig getStorageConfig() {
+    Configuration.StorageConfig storageConfig = config.storageConfig;
 
-  JsonObject getBufferPoolConfig() {
-    JsonObject poolConfig = config.getJsonObject(BUFFER_POOL_OPTIONS_KEY, new JsonObject());
-    if (poolConfig.getInteger("max.total") <= 0 || poolConfig.getInteger("max.idle") < 0 || poolConfig.getInteger("buffer.size") <= 0) {
-      throw new RuntimeException("buffer pool config is not proper");
+    // check for consistent config
+    Long requestTimeout = getUserapiHttpConfig().requestTimeout;
+    Long ListObjectTimeout = storageConfig.s3Config.listObjectsTimeoutMs;
+    if(requestTimeout <= ListObjectTimeout) {
+      throw new RuntimeException("request timeout must be greater than listObject timeout");
     }
-    return poolConfig;
-  }
 
-  private void checkNotEmpty(JsonObject json, String tag) {
-    if (json == null || json.isEmpty()) {
-      // TODO: convert these to configException
-      throw new RuntimeException(tag + " config is not present");
-    }
+    return storageConfig;
   }
 
-  public JsonObject getUserapiHttpDeploymentConfig() {
-    JsonObject deploymentConfig = enrichDeploymentConfig(config.getJsonObject(USERAPI_HTTP_DEPLOYMENT_OPTIONS_KEY, new JsonObject()));
-    JsonObject httpConfig = deploymentConfig.getJsonObject("config");
-    if(!httpConfig.containsKey("req.timeout")) {
-      // put default value for request timeout
-      httpConfig.put("req.timeout", 10000);
-    }
-    return deploymentConfig;
+  public DeploymentOptions getUserapiHttpDeploymentConfig() {
+    return config.httpVerticleConfig;
   }
 
   int getProfileRetentionDuration() {
-    return config.getInteger(PROFILE_RETENTION_KEY, 30);
+    return config.profileRetentionDurationMin;
   }
 
   public int getAggregationWindowDurationInSecs() {
-    return config.getInteger("aggregation_window.duration.secs", 30);
+    return config.aggregationWindowDurationSec;
   }
 
   public String getBaseDir() {
-    return config.getString("base.dir", "profiles");
+    return config.baseDir;
   }
 
-  public int getUserapiHttpPort() {
-    return getUserapiHttpDeploymentConfig().getJsonObject(CONFIG).getInteger(USERPAI_HTTP_PORT_KEY, 8082);
+  public Configuration.HttpConfig getUserapiHttpConfig() {
+    return getUserapiHttpDeploymentConfig().getConfig().mapTo(Configuration.HttpConfig.class);
```

Comment:
Seems a duplicate effort. We serialize json to deployment options and then here we extract json from deployment options again. Maybe have another field in Configuration class to store HttpConfig?
If validations are present in http config, we can also choose to expose only the json (or deployment options wrapper). The other can always be generated